### ISSUE TITLE
Stopping fullsync may not fully stop full sync

### ISSUE THREAD

**Issue by lordnull** [basho/riak_repl#246]
There exists a case where issuing "riak-repl fullsync stop <clustername>" may leave orphaned full sync source processes.

When Issuing a fullsync stop (or disable), the fscoordinator will send a stop message to the fssource. The source will then send a stop to it's worker process, a riak_repl_keylist_server. Under most conditions, the keylist_server will send a stop to it's helper. The helper will then exit with reason 'normal'. Since each process is linked to the next in a chain, the exit will propagate up through the keylist server and the source.

While the keylist server is in certain states, such as 'wait_for_partition', this stop message will not actually do anything, leaving the source to be idle.

A full test and inspection would be required to find the conditions where fullsync processes could be left as a zombie. If a quick fix is required, altering the stop_fullsync handling clause in riak_repl2_fssource.erl may provide sufficient coverage:

```
handle_call(stop_fullsync, _From, State=#state{fullsync_worker=FSW}) ->
    riak_repl_keylist_server:cancel_fullsync(FSW),
-    {reply, ok, State};
+   {stop, normal, ok, State};
```

This would make the exit travel downward.

**Comment by jonmeredith** [comment]
Nice find - we definitely saw this on Thursday night with the customer.

**Comment by cmeiklejohn** [comment]
Moving to RC tag with resilience group tag for work on qualifying 2.0.

**Comment by cmeiklejohn** [comment]
Assigning @cmeiklejohn.

**Comment by macintux** [comment]
In the interest of prioritizing workload, does it make sense to implement the quick fix for 2.0 to limit exposure to the problem and save the more thorough inspection for 2.1, and/or set aside the whole issue for 2.1?

**Comment by cmeiklejohn** [comment]
I think the following pull request addresses the issue: https://github.com/basho/riak_repl/pull/591

Re-open if that is not the case.
